example/lib/sensor_tag_test_scenario.dart

```diff
@@ -52,12 +53,24 @@ class TestScenario {
         log("${peripheral.name} has DISCONNECTED");
       }
     });
-
     log("CONNECTING to ${peripheral.name}, ${peripheral.identifier}...");
     await peripheral.connect();
     log("CONNECTED to ${peripheral.name}, ${peripheral.identifier}!");
     deviceConnectionAttempted = false;
 
+    monitoringStreamSubscription?.cancel();
+    monitoringStreamSubscription = peripheral
```

Comment:
Maybe I do not know Dart enough but it looks weird. Is `Peripheral` a `StreamSubscription`?

Reply:
No. peripheral.monitorCharacteristic returns a stream and Stream.listen() returns a subscription

Reply:
> Maybe I do not know Dart enough but it looks weird. Is `Peripheral` a `StreamSubscription`?

This is a chainging of calls. `peripheral.monitorCharacteristic().listen()` - last call returns a stream subscription.